spec/zoneddatetime.html

```diff
@@ -0,0 +1,1049 @@
+    <ul>
+      <li>is the intrinsic object <dfn>%Temporal.ZonedDateTime.prototype%</dfn>.
+      <li>is itself an ordinary object.
+      <li>is not a Temporal.ZonedDateTime instance and does not have a [[InitializedTemporalZonedDateTime]] internal slot.
+      <li>has a [[Prototype]] internal slot whose value is %Object.prototype%.
+    </ul>
+
+    <emu-clause id="sec-temporal.zoneddatetime.prototype.constructor">
+      <h1>Temporal.ZonedDateTime.prototype.constructor</h1>
+      <p>The initial value of `Temporal.ZonedDateTime.prototype.constructor` is %Temporal.ZonedDateTime%.</p>
+    </emu-clause>
+
+    <emu-clause id="sec-temporal.zoneddatetime.prototype-@@tostringtag">
+      <h1>Temporal.ZonedDateTime.prototype[ @@toStringTag ]</h1>
+      <p>
+        The initial value of the @@toStringTag property is the string value `"Temporal.ZonedDateTime"`.
+      </p>
+      <p>
+        This property has the attributes { [[Writable]]: *false*, [[Enumerable]]: *false*, [[Configurable]]: *true* }.
+      </p>
+    </emu-clause>
+
+    <emu-clause id="sec-get-temporal.zoneddatetime.prototype.calendar">
+      <h1>get Temporal.ZonedDateTime.prototype.calendar</h1>
+      <p>
+        `Temporal.ZonedDateTime.prototype.calendar` is an accessor property whose set accessor function is *undefined*.
+        Its get accessor function performs the following steps:
+      </p>
+      <emu-alg>
+        1. Let _zonedDateTime_ be the *this* value.
+        1. Perform ? RequireInternalSlot(_zonedDateTime_, [[InitializedTemporalZonedDateTime]]).
+        1. Return _zonedDateTime_.[[Calendar]].
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-get-temporal.zoneddatetime.prototype.timezone">
+      <h1>get Temporal.ZonedDateTime.prototype.timeZone</h1>
+      <p>
+        `Temporal.ZonedDateTime.prototype.timeZone` is an accessor property whose set accessor function is *undefined*.
+        Its get accessor function performs the following steps:
+      </p>
+      <emu-alg>
+        1. Let _zonedDateTime_ be the *this* value.
+        1. Perform ? RequireInternalSlot(_zonedDateTime_, [[InitializedTemporalZonedDateTime]]).
+        1. Return _zonedDateTime_.[[TimeZone]].
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-get-temporal.zoneddatetime.prototype.era">
+      <h1>get Temporal.ZonedDateTime.prototype.era</h1>
+      <p>
+        `Temporal.ZonedDateTime.prototype.era` is an accessor property whose set accessor function is *undefined*.
+        Its get accessor function performs the following steps:
+      </p>
+      <emu-alg>
+        1. Let _zonedDateTime_ be the *this* value.
+        1. Perform ? RequireInternalSlot(_zonedDateTime_, [[InitializedTemporalZonedDateTime]]).
+        1. Let _timeZone_ be _zonedDateTime_.[[TimeZone]].
+        1. Let _instant_ be ? CreateTemporalInstant(_zonedDateTime_.[[Nanoseconds]]).
+        1. Let _temporalDateTime_ be ? GetTemporalDateTimeFor(_timeZone_, _instant_).
```

Comment:
General question: it's possible that re-building the DateTime for each property access could get quite expensive, esp in cases where you pull multiple props from the same instance which is common.  Seems like an implementation may want to cache the DateTime the first time it's generated.  Is that something that ECMAScript specs address?  Or is that an implementation detail that the implementers worry about but the spec doesn't?

Reply:
I planned to find out later on what could be done to address this, so I will definitely follow up later, but my understanding is that it's not possible to cache the DateTime in the case of a custom time zone. For example, you could do this:
```
class MyTimeZone extends Temporal.TimeZone {
    constructor() { super('bad'); }
    getDateTimeFor() { return Temporal.DateTime.from('1970-01-01'); }
}
const tz = new MyTimeZone();
const zdt = new Temporal.ZonedDateTime(0n, tz);
zdt.year // => 1970
MyTimeZone.prototype.getDateTimeFor = function() { return Temporal.DateTime.from('2000-01-01'); }
zdt.year // => 2000
```
If we wrote in the spec that the DateTime had to be cached, then you could get into the situation where
the result of `zdt.year` is not equal to `zdt.timeZone.getDateTimeFor(new Instant(zdt.epochNanoseconds)).year`.

Implementations can cache the calls if the time zone is a built-in one because then the call to getDateTimeFor won't be observable.

Reply:
Got it. As long as it's on your radar, that's good enough for me!